### ISSUE TITLE
no-throw-literal gives false positive for objects built using object spread

### ISSUE THREAD

**Issue by danny-andrews** [eslint/eslint#8751]
**Tell us about your environment**

* **ESLint Version:**
v3.19.0
* **Node Version:**
v7.10.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

```yaml
rules:
  no-throw-literal: error
```

**What did you do? Please include the actual source code causing the issue.**

```js
throw { ...new Error('Error loading Instagram Photos'), ...err };
```

**What did you expect to happen?**
No error, I **am** throwing an object.

**What actually happened? Please include the actual, raw output from ESLint.**
`Expected an object to be thrown. (no-throw-literal)`

**Comment by not-an-aardvark** [comment]
This actually isn't related to spread syntax -- it's just that the error message is misleading. By design, the `no-throw-literal` reports when objects are thrown, because those objects are not instances of `Error`.

**Comment by danny-andrews** [comment]
oic, that makes sense. I'll close this. Thank you.
